provider/datagen/README.md

```diff
@@ -80,7 +80,7 @@ $ icu4x-datagen config.json
 
```

Comment:
Did you fix this one?

> DatagenProvider: ExportableProvider link is broken

Reply:
Couldn't reproduce this

Reply:
It's on this page: https://unicode-org.github.io/icu4x/docs/icu_datagen/struct.DatagenProvider.html

The link to ExportableProvider is a 404.

Reply:
It doesn't reproduce locally, so shouldn't be an actual issue on crates.io.